src/screens/transaction-request/transaction-request.tsx

```diff
@@ -79,7 +87,21 @@ export class TransactionRequestScreenComponent extends React.Component<
             const payload = await ConnectExtensionWeb.getRequestIdParams(this.props.requestId);
 
             if (payload) {
-                this.setState({ moonletTransferPayload: payload });
+                const walletId = payload.walletId;
+                if (walletId !== this.props.selectedWallet.id) {
+                    if (
+                        Object.keys(this.props.wallets).filter(wId => wId === walletId).length === 1
+                    ) {
+                        // Switch the wallet
+                        this.props.setSelectedWallet(walletId);
```

Comment:
What happens if on the new selected wallet the selected blockchain is different?

Reply:
It's safe because the `blockchain` is taken from the `account` which is send via transfer payload.
It's not based on the phone's selected blockchain.